Formula/clash.rb

```diff
@@ -0,0 +1,44 @@
+class Clash < Formula
+  desc "Rule-based tunnel in Go"
+  homepage "https://github.com/Dreamacro/clash"
+  url "https://github.com/Dreamacro/clash/archive/v1.3.0.tar.gz"
+  sha256 "34a31e12eb7e639bdc2e19421fe3a8d9ad710e7d4c13bfc361c2023c9dfe04f6"
+  license "GPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  plist_options manual: "#{HOMEBREW_PREFIX}/opt/clash/bin/clash"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+            <array>
+                <string>#{opt_bin}/clash</string>
+            </array>
+            <key>RunAtLoad</key>
+            <true/>
+            <key>KeepAlive</key>
+            <true/>
+            <key>StandardOutPath</key>
+            <string>#{var}/log/clash.log</string>
+            <key>StandardErrorPath</key>
+            <string>#{var}/log/clash.log</string>
+          </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/clash", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for your advices.
I adapt the test of shadowsocks-libev.